ext/sqlite3/sqlite3.c

```diff
@@ -1393,6 +1393,85 @@ PHP_METHOD(sqlite3stmt, readOnly)
 }
 /* }}} */
 
+/* bind parameters to a statement before execution */
+static int php_sqlite3_bind_params(php_sqlite3_stmt *stmt_obj) /* {{{ */
```

Comment:
@nikic thanks, but I'm not sure if returning zend_bool is appropriate? as this is an internal function, only called by other functions in that module, it's not returning directly to the php code. In fact I based this on other functions already present in the source, doing the same thing.

Is there a place where I can find a guide/doc about these details so that I can make my code better? cheers :)

Reply:
zend_bool is just a custom type to work around the fact that C (prior to C99) does not have a native bool type.

Looking at this file, I see there are a lot of functions returning int, but they use different conventions. Most seem to use "0" as a success value, which corresponds to the SUCCESS/FAILURE convention.

Unfortunately I can't point you at any docs for this, it's just a loose convention to distinguish whether the return value is 1/0 or 0/-1 for succes/failure respectively.

Reply:
Makes sense, thanks. Updated the code, will try to remember that next time :)